contracts/HATVaultsRegistry.sol

```diff
@@ -336,22 +341,27 @@ contract HATVaultsRegistry is IHATVaultsRegistry, Ownable {
     ) external onlyOwner {
         // Needed to avoid a "stack too deep" error
         SwapData memory swapData;
-        swapData.amount = governanceHatReward[_asset];
-        for (uint256 i = 0; i < _beneficiaries.length; i++) {
-            swapData.amount += hackersHatReward[_asset][_beneficiaries[i]];
+        swapData.hackerRewards = new uint256[](_beneficiaries.length);
+        swapData.governanceHatReward = governanceHatReward[_asset];
+        swapData.amount = swapData.governanceHatReward;
+        for (uint256 i = 0; i < _beneficiaries.length;) { 
+            swapData.hackerRewards[i] = hackersHatReward[_asset][_beneficiaries[i]];
+            swapData.amount += swapData.hackerRewards[i]; 
+            unchecked { ++i; }
         }
         if (swapData.amount == 0) revert AmountToSwapIsZero();
         IERC20 _HAT = HAT;
         (swapData.hatsReceived, swapData.amountUnused) = _swapTokenForHAT(IERC20(_asset), swapData.amount, _amountOutMinimum, _routingContract, _routingPayload);
         
-        swapData.governanceAmountSwapped = (swapData.amount - swapData.amountUnused).mulDiv(governanceHatReward[_asset], swapData.amount);
-        governanceHatReward[_asset] = swapData.amountUnused.mulDiv(governanceHatReward[_asset], swapData.amount);
+        swapData.usedPart = (swapData.amount - swapData.amountUnused) / swapData.amount;
+        swapData.governanceAmountSwapped = swapData.usedPart * swapData.governanceHatReward;
```

Comment:
Hmm this gives up the mulDiv use.

Reply:
hmm right.. maybe better to have it with mulDiv actually